core/test/refx/subs_test.cljs

```diff
@@ -104,3 +104,39 @@
                               (is (empty? @subs/sub-cache))
                               (done))
                             10)))))
+
+(deftest listener-ordering
+  (let [source (atom 0)]
+    (subs/register :a (constantly source) identity)
+    (subs/register :b (constantly source) identity)
+    (subs/register :c (constantly (subs/sub [:b (subs/sub [:a])])) identity)
+    (subs/register :d (constantly [(subs/sub [:b]) (subs/sub [:c])]) identity)
+    (let [sub-a (subs/sub [:a])
+          sub-b (subs/sub [:b])
+          sub-c (subs/sub [:c])
+          sub-d (subs/sub [:d])
+          listener-count (atom 0)
+          listener-calls (atom '())
+          remove-listener-fns (atom '())
+          add-listener! (fn [sub]
+                          (let [key {:index (swap! listener-count inc)}]
+                            (subs/-add-listener sub key #(swap! listener-calls conj key))
+                            (swap! remove-listener-fns conj #(subs/-remove-listener sub key))))
+          remove-listeners! (fn []
+                              (doseq [f @remove-listener-fns]
+                                (f)))]
+      (add-listener! sub-a)
+      (add-listener! sub-b)
+      (add-listener! sub-c)
+      (add-listener! sub-d)
+      (reset! source 1)
+      (remove-listeners!)
+      (async done
+             (js/setTimeout (fn []
+                              (is (= @listener-calls
```

Comment:
Without these changes, this test fails and listener-calls has a value of:

```clj
({:index 2}
 {:index 4}
 {:index 1}
 {:index 3}
 {:index 4})
```

Note that order isn't guaranteed and 4 occurs twice.